### PR TITLE
Add FXIOS-2912 [v100]:  STG in Recently Visited

### PR DESCRIPTION
**Pull request by nishant2718**
# Overview

PR in reference to https://mozilla-hub.atlassian.net/browse/FXIOS-2912.

### CODE REVIEW

**Comment by nishant2718** [comment]
# Request for data collection review form

All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.

1. What questions will you answer with this data?
- How many users tap a grouping inside Recently Visited on Firefox Homepage. 
- How many users tap on a highlight item in Recently Visited on Firefox Homepage. 

2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?
  * This will help us understand usage patterns around the recently visited section. 
3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
  * There are no other alternatives.
4. Can current instrumentation answer these questions?
  * No
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.
Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
The highlight item was clicked | Category 2 - interaction data | https://mozilla-hub.atlassian.net/browse/FXIOS-2912
A grouped item type was clicked | Category 2 - interaction data | https://mozilla-hub.atlassian.net/browse/FXIOS-2912

6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
  * https://dictionary.telemetry.mozilla.org/apps/firefox_ios
7. How long will this data be collected? Choose one of the following:
  * This will expire at the end of the year.
8. What populations will you measure?
  * All channels, all locales, all countries
9. If this data collection is default on, what is the opt-out mechanism for users?
  * Default Glean SDK opt-out mechanism.
10. Please provide a general description of how you will analyze this data.
  * Glean
11. Where do you intend to share the results of your analysis?
  * Only on Glean, and with mobile teams.
12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
   * No third-party tools.

**Comment by dnarcese** [comment]
r+
# Data Review Form

_1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?_ 

Yes, telemetry is documented at https://dictionary.telemetry.mozilla.org/apps/firefox_ios

_2) Is there a control mechanism that allows the user to turn the data collection on and off?_ 

Yes, Firefox iOS provides data controls in the app settings

_3) If the request is for permanent data collection, is there someone who will monitor the data over time?_

This should be re-reviewed w/ Ashley Thomas every 6mo

_4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?_

Type 2

_5) Is the data collection request for default-on or default-off?_

Default on

_6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?_

No

_7) Is the data collection covered by the existing Firefox privacy notice?_ 

Yes

_8) Does the data collection use a third-party collection tool?_ 

No

**Comment by nbhasin2** [comment]
I am seeing empty screenshot or white in recently saved, is that normal or expected?

image

**Comment by lmarceau** [comment]
> I am seeing empty screenshot or white in recently saved, is that normal or expected?
> 
> image

This has been there for a while, it relates to https://mozilla-hub.atlassian.net/browse/FXIOS-3553

File: Client/FeatureFlags/FeatureFlagsManager.swift [hunk 1]
```diff
@@ -146,7 +146,7 @@ class FeatureFlagsManager {
 
         let historyHighlights = FlaggableFeature(withID: .historyHighlights,
                                                  and: profile,
-                                                 enabledFor: [.developer])
+                                                 enabledFor: [.release, .beta, .developer])
```
Comment (nbhasin2) [inline Client/FeatureFlags/FeatureFlagsManager.swift:149]: we are good to put it in release? 

do we need this behind nimbus flag and if so do we need to do anything in the new nimbus layer?

Comment (nbhasin2) [reply Client/FeatureFlags/FeatureFlagsManager.swift:149]: cc @nishant2718 @yoanarios

**Review by nbhasin2 — APPROVED** [review]

**Comment by nbhasin2** [comment]
> > I am seeing empty screenshot or white in recently saved, is that normal or expected?
> > image
> 
> This has been there for a while, it relates to https://mozilla-hub.atlassian.net/browse/FXIOS-3553

Ah gotcha, should look into this as there is a fallback and instead of being empty it should go to favicon 🤔 

This should not be a blocker for PR but something to address in future

File: Client/FeatureFlags/FeatureFlagsManager.swift (hunk 1 above)
Comment (yoanarios) [reply Client/FeatureFlags/FeatureFlagsManager.swift:149]: @nbhasin2 yes Daniela ask us to enable for release. As for the nimbus flag I'm not sure if we need something else in the `FxHomeHistoryHightlightsViewModel` `isEnabled` I see only the check for `isFeatureActiveForBuild` and `userPreferenceFor`. What do you think?

Comment (nbhasin2) [reply Client/FeatureFlags/FeatureFlagsManager.swift:149]: I’d recommend rebasing from main and making sure nimbus layer has all the required changes for your .isEnabled flag. Let me find the example @yoanarios

Comment (nbhasin2) [reply Client/FeatureFlags/FeatureFlagsManager.swift:149]: cc @electricRGB

**Pull request merged**